[cjyothika] iP

cjyothika opened the pull request:
# Duke
> Why remember things? You're not going to do them anyway ~ Someone probably

Well, Duke is going to bring you closer to your goals, I guess *(still not completing them though)*. It's
- going to repeat what you tell it
- repeat what you say to it
- more repetition so I guess that helps, right?

All you need to do is,
1. download a release dummy (https://github.com/nus-cs2103-AY2223S2/ip)
2. run the following command in cli
`java -jar -insert file path-`
3. let it repeat your tasks back to you
4. Profit :heavy_dollar_sign:

And it involves **minimal** effort?

## Features:
- [X] repeating tasks back to you
- [ ] was i supposed to think of more features? oops

If you're a Programmer, I guess here's some code to read:
```
System.out.println("Hello world");
```

Commit:
Add Gradle support

Commit:
Level-1: added code to greet user, accept input and bid farewell to the user and added .class files to gitignore file

Commit:
Level-2: Add, List functionalities added

Commit:
Duke.java: added mark and unmark functionalities for tasks
Task.java: created task class to store if a task a marked and its description

Commit:
Duke.java: added 3 different types of tasks and their functionalities
Deadline.java: added deadline class
Event.java: added event class
Todo.java: added todo class

Commit:
EXPECTED.TXT: added expected output
input.txt: added test commands

Commit:
Duke.java: Added functions to handle checked exceptions
Expected.txt: updated intended test results of new tests
input.txt: added new tests for exception handling
DukeException.java: added new Exception class

Commit:
Duke.java: added delete task function and replaced task array with arraylist
input.txt: added more tests for delete function
EXPECTED.TXT: added expected behaviour for new tests for delete function

Commit:
Added Enums to Duke.java

Commit:
Add saving tasks to file functionality

Save tasks in a text file 'data/duke.txt' relative to the project root to be retrieved when using Duke again.

Commit:
Add deadlines as LocalDate objects

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Commit:
Add more Classes for greater abstraction

Add Command and children classes, Ui class, Parser Class, Storage Class.

Commit:
Fix spacing bug for Ui

Make spacing between lines of commands consistent

Commit:
Organize classes into packages duke, duke.command and duke.task

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add JUnit tests for Parser and TaskList classes

Commit:
Fix bugs in Parser class and Storage class

Edit code for load function in Storage class to create directories if they did not previously exist

Commit:
Add JavaDocs for all class files

Commit:
Tweak code to comply with coding standards

Commit:
Add find command function

Add FindCommand class and functionality to find tasks with substrings

Commit:
Merge branch 'branch-Level-9'

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/command/Parser.java
#	src/main/java/duke/task/TaskList.java

Commit:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	data/tasks.txt
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/ExitCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/MarkCommand.java
#	src/main/java/duke/command/Parser.java
#	src/main/java/duke/command/UnknownCommand.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/TaskList.java
#	src/main/java/duke/task/Todo.java

yaojiax commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,63 @@
+    /**
+     * Run.
+     */
+    public void run() {
```
Could possibly use protected here instead of public

cjyothika replied:
True. Thanks for the suggestion. However, I was considering the case where a user might want to call run() again on an already created Duke object.

yaojiax commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,84 @@
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        switch (this.type) {
+            case "E":
```
switch and case should have the same indentation, i.e. they should start from the same line. No worries, this is a common mistake.

cjyothika replied:
Thanks for this suggestion. I missed that. :)

yaojiax commented on src/test/java/duke/command/ParserTest.java:
```diff
@@ -0,0 +1,40 @@
+    public void parseDeadlineTest() throws DukeException {
+        String[] actual = Parser.processDeadline("deadline hw /by 2023-01-01");
+        String[] expected = new String[]{" hw ", "2023-01-01"};
+        assertEquals(expected.length, actual.length);
```
It's clear as to what you are testing. Nice!

yaojiax commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,36 @@
+public class MarkCommand extends Command{
+
+    private int index;
+    private boolean isMarked;
```
Good naming of Boolean variables!

yaojiax reviewed (commented):
Nice and readable code with just a few coding style violation. Keep it up!

Commit:
Add GUI for Duke in DukeView and Launcher classes

Commit:
Merge branch 'branch-Level-10'

Cheamybunny commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,128 @@
+ */
+public class Parser {
+
+    public enum Keyword {
```
Good use of ENUMS

Cheamybunny reviewed (commented):
Looks good so far, look forward to looking at your GUI  implementation

Commit:
Fix bugs for basic GUI

Handle errors by showing them on the GUI as a response to the User

Commit:
Update gitignore to include txt files in data directory

Commit:
Update gitignore to fix bug of data txt files being staged

Commit:
Add CheckStyle

Add checkstyle config files and add checkstyle dependencies to build.gradle file

Commit:
Merge branch 'branch-A-CheckStyle'

Commit:
Update build.gradle to run Launcher as main class

Run GUI instead of CLI interface when duke is ran

Commit:
Add assertions to TaskList class

Exceptions are handled in the AddCommand classes. Ensure that these Exceptions are caught and the appropriate task number is passed to TaskList functions

Commit:
Add comments for lines with abstracted functions

There are many abstractions of functions in separate classes. Add the description of abstracted function to explain why function is applied

Commit:
Add streams to find function of TaskList.

There is manually array iteration used to match search word to every stored task.

Streams can be used for this task so that the code is more concise.

cjyothika committed:
Merge pull request #1 from cjyothika/branch-A-Assertions

Add assertions to the TaskList class

To ensure correct input is provided for mark, unmark and delete functions in the TaskList class.

Correct input for these functions should be a valid task number.

Commit:
Merge branch 'master' of https://github.com/cjyothika/ip into branch-A-Streams

Update Stream with changes from A-Assertions branch to master

Commit:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/command/DeleteCommand.java
Include both comment and code change to throw exception when invalid task index is passed to DeleteCommand on execute function use

cjyothika committed:
Merge pull request #2 from cjyothika/branch-A-Streams

Add streams to TaskList find function

Use streams to simplify existing code that uses manual ArrayList iteration.

cjyothika committed:
Merge pull request #3 from cjyothika/branch-A-CodeQuality

Improve code quality

Add more comments to Parser class to explain why certain functions are called on the user input to extract key information.

Commit:
Add C-Statistics extension

Add "stats" command that returns number of done and undone tasks at the moment.

Do this through the new StatsCommand class which calls the stats function of a TaskList object to return the required information in a string format.

Commit:
Merge branch 'branch-BCD-Extension'

Commit:
Change GUI to make it more colourful and personalised

Add EggDuke themed profile pictures and wallpaper

Add speech bubble-like dialog containers by giving them a black border

Commit:
Fix bug related to GUI where some text is cutoff by the next dialog container

Let's enable the wrapText property to do so.

Commit:
Merge branch 'branch-A-BetterGui'

Commit:
Catch more exceptions and throw them with a message that
user can understand

There are exceptions caught that exposes code to User such as
invalid enum.

This violates information hiding and abstraction.

Let's catch these exceptions and display them to user in a
user-friendly manner such as "I don't know what that means".

Commit:
Merge branch 'A-MoreErrorHandling'

Commit:
Add text emojis to messages to add personality to Eggduke

Emojis add a personal feel

Commit:
Add title to EggDuke application

Let's set the title to the window generated by JavaFx.

Commit:
Add black border to text input box

Commit:
Merge branch 'branch-A-Personality'

Commit:
Add respresentative screenshot to docs folder

Finalize features

Commit:
Update README.md

Add instructions on how to set-up and use EggDuke for users

cjyothika committed:
Update README.md

Duplicate README.md in docs folder

cjyothika committed:
Update README.md

Correct formatting error in header

Set-up instructions header is at wrong level.

Commit:
Add test for find function in TaskList class and edit javadocs to
fit conventions

Extra test using the same test case for add and delete functions is
used to check if find function is working appropriately as well

Some header comments are not using the correct verbs which makes
Javadocs comments inconsistent.

Let's attempt to make all comments
uniform.

Commit:
Merge branch 'master' of https://github.com/cjyothika/ip

* 'master' of https://github.com/cjyothika/ip:
  Update README.md
  Update README.md

Commit:
Update User guide to reflect correct jar file name to avoid
confusion

Release is called duke.jar but user guide refers to release as
Duke.jar.

Let's keep the name of the release as duke.jar in both places.

Commit:
Update User guide to reflect correct jar file name to avoid
confusion

Release is called duke.jar but user guide refers to release as
Duke.jar.

Let's keep the name of the release as duke.jar in both places.

Commit:
Change Ui.png to a clearer image

Image is a bit blurry. UI looks not very polished.

Let's use a better quality image instead.

Commit:
Change Ui.png to a clearer image

Image is a bit blurry. UI looks not very polished.

Let's use a better quality image instead.
